<launch>
  <!-- nodelet manager process -->
  <node pkg="nodelet" type="nodelet" name="nodelet_manager" output="screen"
        args="manager" />

  <!-- launch the openni kinect node -->


  <!-- indoor slam nodelet -->
  <node pkg="nodelet" type="nodelet" name="idslam_nodelet" output="screen"
	args="load idSLAM/idslamnodelet nodelet_manager" >
	<remap from="rgb" to="/camera/rgb/image_color" />
        <remap from="depth" to="/camera/depth_registered/image_raw" />
	<remap from="rgb1" to="/camera0/rgb/image_color" />
        <remap from="depth1" to="/camera0/depth_registered/image_raw" />
	<param name="ini_method" type="string" value="kinect" />

        <param name="usingdualslam" type="bool" value="false" />
        <param name="sendvisual" type="bool" value="true" />
        <param name="show_debug_image" type="bool" value="true" />
	<param name="world_frame" type="string" value="/camera_link" />
        <param name="write_pos" type="bool" value="true" />
        <param name="use_artificial_ini" type="bool" value="false" />
        <param name="use_ekf" type="bool" value="false" />
        <param name="flying" type="bool" value="true" />
 	
        <param name="trackingpad" type="bool" value="false" />
        
	<param name="cam_imu_file" type="string" value="$(find idSLAM)/data/parameter_wrt.txt" />
        <param name="cam_sec_imu_file" type="string" value="$(find idSLAM)/data/kinectcam1cam2.txt" />

        <param name="ref_img_path" type="string" value="$(find idSLAM)/data/refpattern12.jpg" />>
        <!--param name="ref_img_path" type="string" value="$(find idSLAM)/data/box212.jpg" /-->

        <param name="settings_file" type="string" value="$(find idSLAM)/data/settings-kinect.cfg" />
        <param name="camera_type" type="string" value="Polynomial"/>
        <param name="camera_file" type="string" value="$(find idSLAM)/data/kinect-poly.xml" />
        <param name="camerasec0_type" type="string" value="Polynomial"/>
        <param name="camerasec0_file" type="string" value="$(find idSLAM)/data/kinect-poly0.xml" />
        <param name="camerasec1_type" type="string" value="Polynomial"/>
        <param name="camerasec1_file" type="string" value="$(find idSLAM)/data/kinect-poly1.xml" />


	<param name="camera_frame" type="string" value="/openni_rgb_optical_frame" />	
	<param name="calib_file" type="string" value="$(find idSLAM)/data/kinect.yaml"/>

        <param name="vocabulary_file" type="string" value="$(find idSLAM)/data/indoor_voc.yml.gz"/>
        <param name="save_keyframes" type="bool" value="false"/>
        <param name="close_loops" type="bool" value="true"/>

  </node>
</launch>
